### PR TITLE
Linux support

### PR DESCRIPTION
**Pull request by melvyn2**
### Related Issue
#2 

### Implementation
Adds build scripts to automate the building process
Fixes some VPC scripts
Changes some code to make it cross-platform 

### Screenshots
Screenshot_2021-08-12_21-57-45

### Checklist

- [x] I certify that this PR is my own entirely original work, or I have permission from and have attributed the relevant authors.
- [x] I have agreed to and signed this project's [Contributor License Agreement](https://cla-assistant.io/mastercomfig/team-comtress-2).
- [x] This PR only contains changes to the engine and/or core game framework
- [ ] This PR targets the `community` branch.

- [x] This change has been filed as an issue.
- [ ] No other PRs address this.
- [x] This PR is as minimal as possible.
- [x] This PR does not introduce any regressions.
- [x] This PR has been built and locally tested on at least one platform.
- [ ] This PR has been tested on all platforms, on both a dedicated server and on a listen server.

### Testing Checklist

|         |            Client             |            Server             | Version                     |
|---------|:-----------------------------:|:-----------------------------:|-----------------------------|
| Windows | TBT | TBT |     |
|   Linux | Tested | N/A | `5.13.10-xanmod1-cacule-1 #1 SMP PREEMPT Fri, 13 Aug 2021 01:16:53 +0000` |
|  Mac OS | N/A | N/A |       |

### Caveats
dunno

### Alternatives
no linux

### CODE REVIEW

**Comment by melvyn2** [comment]
@HurricanePootis it is real

File: src/launcher/launcher.cpp [hunk 1]
```diff
@@ -647,13 +647,13 @@ bool CSourceAppSystemGroup::Create()
 
 #ifdef WIN32
 	CoInitialize( NULL );
+	ThreadSpin(1);
```
Comment (mastercoms) [inline src/launcher/launcher.cpp:650]: Has to happen after Plat_FloatTime.

File: src/public/tier1/utltshash.h [hunk 2]
```diff
@@ -617,7 +617,6 @@ inline UtlTSHashHandle_t CUtlTSHash<T,BUCKET_COUNT,KEYTYPE,HashFuncs,nAlignment>
 {
 	Assert( pElement );
 	HashFixedData_t *pFixedData = (HashFixedData_t*)( (uint8*)pElement - offsetof( HashFixedData_t, m_Data ) );
-	Assert( m_EntryMemory.IsAllocationWithinPool( pFixedData ) );
```
Comment (mastercoms) [inline src/public/tier1/utltshash.h:620]: Why is this removed?

File: src/tier0/threadtools.cpp [hunk 3]
```diff
@@ -267,6 +267,11 @@ unsigned int g_optimalMaxNormalizedYieldsPerSpinIteration = 7;
 const unsigned int MinNsPerNormalizedYield = 37; // measured typically 37-46 on post-Skylake
 const unsigned int NsPerOptimalMaxSpinIterationDuration = 272; // approx. 900 cycles, measured 281 on pre-Skylake, 263 on post-Skylake
 
+#ifdef POSIX
+bool s_isYieldProcessorNormalizedInitialized = false;
```
Comment (mastercoms) [inline src/tier0/threadtools.cpp:271]: I would rather you do an implementation of Init shich just sets s_isYieldProcessorNormalizedInitialized to true, and add a todo.

**Review by mastercoms — CHANGES_REQUESTED** [review]

File: src/public/tier1/utltshash.h (hunk 2 above)
Comment (melvyn2) [reply src/public/tier1/utltshash.h:620]: The function `IsAllocationWithinPool` doesn't exist.

File: src/tier0/threadtools.cpp (hunk 3 above)
Comment (melvyn2) [reply src/tier0/threadtools.cpp:271]: I meant to set it to true, will change as requested.

File: src/public/tier1/utltshash.h (hunk 2 above)
Comment (mastercoms) [reply src/public/tier1/utltshash.h:620]: Why can I compile it now?

Comment (melvyn2) [reply src/public/tier1/utltshash.h:620]: Probably something about how MSVC handles asserts, I know I had to remove this to get GCC or clang to compile it.

Comment (mastercoms) [reply src/public/tier1/utltshash.h:620]: Can you comment it as a TODO? It's present in later version of mempool.

**Review by mastercoms — APPROVED** [review]
